dotenv_linter/violations/base.py

```diff
@@ -27,7 +28,9 @@ def as_line(self) -> str:
 
     def location(self) -> int:
         """Returns in-file location of a violation."""
-        raise NotImplementedError('Should be redefined in a subclass')
+        with pytest.raises(NotImplementedError):
```

Comment:
That should be placed inside a test, not in the implementation. See this example.
Test: https://github.com/wemake-services/wemake-python-styleguide/blob/master/tests/test_visitors/test_base.py#L14-L17
Source: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/visitors/base.py#L129